Sources/Routing/Register/PathComponent.swift

```diff
@@ -1,11 +1,68 @@
+import Foundation
+
+/// A type that can be created from `Data` in a lossless, unambiguous way.
+public protocol LosslessDataConvertible {
+    /// Losslessly converts `Data` to this type.
+    init?(_ data: Data)
+}
+
+public extension LosslessDataConvertible {
+    init?(_ data: Data) {
+        guard data.count < MemoryLayout<Self>.size else {
```

Comment:
This bit is hella janky and I'm hoping I or someone else can come up with a better means of failing early. The point is that if we see evidence that the input data isn't able to be converted to the expected type, we should fail, and we should do that as early as possible for performance purposes. 

The suggested means of doing this online is checking if `data.count == MemoryLayout<ExpectedType>.size`, but in practice this doesn't work at all. For example in the case of a request to `/foo/1` where we expect the second parameter to be an `Int` "1", despite being a perfectly viable `Int` is only 1 byte, and  `MemoryLayout<ExpectedType>.size` is 8, so the above boolean is false. I'm not sure how to remedy this, but I'm super open to suggestions.